manifests/init.pp

```diff
@@ -442,7 +442,20 @@
   if $ssh_config_hash_known_hosts_real != undef {
     validate_re($ssh_config_hash_known_hosts_real, '^(yes|no)$', "ssh::ssh_config_hash_known_hosts may be either 'yes' or 'no' and is set to <${ssh_config_hash_known_hosts_real}>.")
   }
-  validate_re($sshd_config_port, '^\d+$', "ssh::sshd_config_port must be a valid number and is set to <${sshd_config_port}>.")
+  case type3x($sshd_config_port) {
+    'string': {
+      validate_re($sshd_config_port, '^\d+$', "ssh::sshd_config_port must be a valid number and is set to <${sshd_config_port}>.")
+    }
+    'array': {
+      validate_array($sshd_config_port)
```

Comment:
Correct me if I'm wrong but do you really need to validate that sshd_config_port is an array if the type3x functions returns that it is an array?

I would propose that we also use validate_numeric on the array to make sure the array only contain values with numbers between 1 to 65525. Looking at stdlibs documentation it should work. https://github.com/puppetlabs/puppetlabs-stdlib#validate_numeric